File: backend/src/main/java/com/yagatalk/domain/Context.java
package com.yagatalk.domain;

import java.time.Instant;
import java.util.UUID;

public class Context {
    private final UUID id;
    private final String content;

    private final Instant createdTime;
    private final String name;

    public Context(UUID id, String content,Instant createdTime, String name) {
        this.id = id;
        this.name = name;
        this.createdTime = createdTime;
        this.content = content;
    }

    public UUID getId() {
        return id;
    }

    public String getContent() {
        return content;
    }

    public String getName() {
        return name;
    }

    public Instant getCreatedTime() {
        return createdTime;
    }
}

File: admin-web-component/src/Context.js
import {useParams} from "react-router-dom";
import React, {useEffect, useState} from "react";

function ContextDetail() {
    const { contextId } = useParams();
    const [contextData, setContextData] = useState(null);

    useEffect(() => {
        const fetchData = () => {
            fetch(`http://localhost:8082/api/chat/sessions/context/${contextId}`)
                .then(response => response.json())
                .then(data => setContextData(data))
                .catch(error => console.error('Error fetching contexts:', error));
        };

        fetchData();
    }, [contextId]);

    return (
        <div className="context-table-page">
            <div className="chat-header">
                YagaTalk
            </div>
            <div className="action-buttons">
                <button className="action-button">Test Chat</button>
                <button className="action-button">Archive Context</button>
            </div>
            {contextData && (
                <div className="context-table">
                    <table>
                        <thead>
                        <tr>
                            <th>Name</th>
                            <th>Created Time</th>
                            <th>Content</th>
                        </tr>
                        </thead>
                        <tbody>
                        <tr>
                            <td>{contextData.name}</td>
                            <td>{contextData.createdTime}</td>
                            <td>{contextData.content}</td>
                        </tr>
                        </tbody>
                    </table>
                </div>
            )}
        </div>
    );
}

export default ContextDetail;